spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -338,6 +338,9 @@ protected void doStop() {
 	protected void actualStart() throws Exception {
 		this.aborted = false;
 		this.hasStopped = false;
+		if (getPrefetchCount() < this.messagesPerAck) {
+			setPrefetchCount(this.messagesPerAck);
```

Comment:
It's fine if it's greater - prefetch simply means how many un-ack'd messages can be outstanding at any time. If prefetch is 15 and mPA is 10, it means we'll have 5 unack'd at the time we send the ack (and then the broker will send 10 more, if available).

If it's smaller, we will be deadlocked with the broker waiting for an ack and us waiting for more messages.